jina/serve/executors/__init__.py

```diff
@@ -112,22 +112,38 @@ class _FunctionWithSchema(NamedTuple):
     def get_function_with_schema(fn: Callable) -> T:
 
         docs_annotation = fn.__annotations__.get('docs', None)
+
         if type(docs_annotation) is str:
             warnings.warn(
                 f'`docs` annotation must be a type hint, got {docs_annotation}'
                 ' instead, you should maybe remove the string annotation. Default value'
                 'DocumentArray will be used instead.'
             )
             docs_annotation = None
+        elif not isinstance(docs_annotation, type):
+            warnings.warn(
+                f'`docs` annotation must be a class if you want to use it'
```

Comment:
Maybe u can add something specific to `Optional` that would be the most common usecase

Reply:
but do we even expect ppl using `Optional` here ? I think this is a relic from the past when we were sending the da to the Executor even if it was empty ?

Reply:
but many old hub Executors are using this, so we do expect people to use old Executors yes

Reply:
ANNLiteIndexer for instance is used and has this behavior

Reply:
I see I see